### ISSUE TITLE
Shutdown sometimes leaves lock files in ensime_cache

### ISSUE THREAD

**Issue by dragos** [dragos/dragos-vscode-scala#2]
It looks like the shutdown sequence isn't correct. We need to make sure every thread cleans up.

**Comment by dragos** [comment]
See ensime/ensime-server#1640

**Comment by jvican** [comment]
Indeed. On a related topic, I've noticed that when quitting vscode the ensime threads do not get killed and survive in the background. I have to kill them manually with htop.

**Comment by jvican** [comment]
This now happens less often than before, and usually after a certain amount of time the server shuts down alone, in my experience.
